brave/brave-core
# feat(wallet): Rewards Custodial Balance in Wallet

## Description 
Introduces `Brave Rewards` custodial balances in the `Brave Wallet`


Resolves <https://github.com/brave/brave-browser/issues/31713>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Testing Connected Rewards:
1. Enable `Rewards` in the `Browser` and link your account with a `Custodial Provider` example: Uphold
2. Open the `Wallet` and go to the `Portfolio` page.
3. You should see your `Custodial` rewards in the `Portfolio`
4. Open the `Filters and display settings` and switch `Group by` between `None, Accounts and Networks`
5. The rewards should display correctly in the `Portfolio`
6. Open the `Filters and display settings` and deselect the `Custodial Provider` under `Platforms`
7. You should not see you `Custodial` rewards in the `Portfolio`
8. Open the `Filters and display settings` and reselect the `Custodial Provider` under `Platforms`
9. You should see you `Custodial` rewards in the `Portfolio`
10. Click on your `Rewards` token, it should open the `Asset Details` view.
11. Test the more menu for the asset and make sure it routes correctly to `Provider Account` or `Rewards Settings`
12. Go to the `Accounts` tab, you should see your `Custodial` rewards as an account.

https://github.com/brave/brave-core/assets/40611140/e09e9eff-345f-4d0d-9b10-7e6c925e0917

## Testing Disconnected Rewards:
1. Go to your `Custodial Provider` account and remove `Brave` from account `Integrations`
2. Go to the Wallet `Portfolio` and refresh the page (Sometimes take a minute to update)
3. The `Custodial Rewards` should no longer be visible
4. Go to the `Accounts` tab, you should see your `Custodial Rewards` account displaying its `Logged out`
5. Click the `Log in` button, it should route you to your `Custodial Provider` account
6. Authorize the `Connection`
7. Go back to the Wallet and refresh the page (Sometimes take a minute to update)
8. It should be visible again in the `Portfolio` and not show `Logged out` on the `Accounts` tab.

https://github.com/brave/brave-core/assets/40611140/5a180366-c841-427f-a0df-6419cc8074e3
